pkg/pillar/cmd/diag/diag.go

```diff
@@ -800,50 +820,174 @@ func tryGetUuid(ctx *diagContext, ifname string) bool {
+		return nil
+	default:
+		return fmt.Errorf("Content-type %s not supported",
+			mimeType)
+	}
+}
+
+func readConfigResponseProtoMessage(contents []byte) (*zconfig.ConfigResponse, error) {
+	var configResponse = &zconfig.ConfigResponse{}
+
+	err := proto.Unmarshal(contents, configResponse)
+	if err != nil {
+		log.Errorf("Unmarshalling failed: %v", err)
+		return nil, err
+	}
+	return configResponse, nil
+}
+
 // Get something without a return type; used by ping
 // Returns true when done; false when retry.
 // Returns the response when done. Caller can not use resp.Body but
 // can use the contents []byte
-func myGet(zedcloudCtx *zedcloud.ZedCloudContext, requrl string, ifname string,
+func myGet(zedcloudCtx *zedcloud.ZedCloudContext, reqURL string, ifname string,
 	retryCount int) (bool, *http.Response, []byte) {
 
-	var preqUrl string
-	if strings.HasPrefix(requrl, "http:") {
-		preqUrl = requrl
-	} else if strings.HasPrefix(requrl, "https:") {
-		preqUrl = requrl
+	var preqURL string
+	if strings.HasPrefix(reqURL, "http:") {
+		preqURL = reqURL
+	} else if strings.HasPrefix(reqURL, "https:") {
+		preqURL = reqURL
 	} else {
-		preqUrl = "https://" + requrl
+		preqURL = "https://" + reqURL
 	}
-	proxyUrl, err := zedcloud.LookupProxy(zedcloudCtx.DeviceNetworkStatus,
-		ifname, preqUrl)
+	proxyURL, err := zedcloud.LookupProxy(zedcloudCtx.DeviceNetworkStatus,
+		ifname, preqURL)
 	if err != nil {
 		fmt.Printf("ERROR: %s: LookupProxy failed: %s\n", ifname, err)
-	} else if proxyUrl != nil {
+	} else if proxyURL != nil {
 		fmt.Printf("INFO: %s: Proxy %s to reach %s\n",
-			ifname, proxyUrl.String(), requrl)
+			ifname, proxyURL.String(), reqURL)
 	}
 	const allowProxy = true
 	resp, contents, rtf, err := zedcloud.SendOnIntf(*zedcloudCtx,
-		requrl, ifname, 0, nil, allowProxy)
+		reqURL, ifname, 0, nil, allowProxy)
 	if err != nil {
 		if rtf {
 			fmt.Printf("ERROR: %s: get %s remote temporary failure: %s\n",
-				ifname, requrl, err)
+				ifname, reqURL, err)
+		} else {
+			fmt.Printf("ERROR: %s: get %s failed: %s\n",
+				ifname, reqURL, err)
+		}
+		return false, nil, nil
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		fmt.Printf("INFO: %s: %s StatusOK\n", ifname, reqURL)
+		return true, resp, contents
+	case http.StatusNotModified:
+		fmt.Printf("INFO: %s: %s StatusNotModified\n", ifname, reqURL)
+		return true, resp, contents
+	default:
+		fmt.Printf("ERROR: %s: %s statuscode %d %s\n",
+			ifname, reqURL, resp.StatusCode,
+			http.StatusText(resp.StatusCode))
+		fmt.Printf("ERRROR: %s: Received %s\n",
+			ifname, string(contents))
+		return false, nil, nil
+	}
+}
+
+func myPost(zedcloudCtx *zedcloud.ZedCloudContext, reqURL string, ifname string,
+	retryCount int, reqlen int64, b *bytes.Buffer) (bool, *http.Response, []byte) {
+
+	var preqURL string
+	if strings.HasPrefix(reqURL, "http:") {
+		preqURL = reqURL
+	} else if strings.HasPrefix(reqURL, "https:") {
+		preqURL = reqURL
+	} else {
+		preqURL = "https://" + reqURL
+	}
```

Comment:
This logic to make sure there is https:// or http:// -- This can be refactored..